test/system/lifecycle/lifecycle_test.go

```diff
@@ -0,0 +1,311 @@
+
+			oidc = &unstructured.Unstructured{}
+			oidc.SetAPIVersion(oidcAPIVersion)
+			oidc.SetKind(oidcKind)
+			oidc.SetName("custom")
+			err = f.ShootClient.Client().Get(ctx, client.ObjectKeyFromObject(oidc), oidc)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Conversion should be safe
+			spec := oidc.Object["spec"].(map[string]interface{})
+			Expect(spec["clientID"]).To(Equal(clientId))
+			Expect(spec["issuerURL"]).To(Equal(oidConfig.Issuer))
+
+			// Get token from seed
+			var ttl int64 = 1800
+			tokenReq, err := f.SeedClient.Kubernetes().CoreV1().ServiceAccounts("default").CreateToken(ctx, "default", &authenticationv1.TokenRequest{
+				Spec: authenticationv1.TokenRequestSpec{
+					Audiences:         []string{clientId},
+					ExpirationSeconds: &ttl,
+				},
+			}, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			expectedStatus := status{
+				Status:  "Failure",
+				Message: "forbidden: User \"custom-prefix:system:serviceaccount:default:default\" cannot get path \"/\"",
+				Reason:  "Forbidden",
+				Code:    403,
+			}
+
+			// Expect API calls to be authenticated
+			Eventually(func() status {
+				statusError, err := requestAPIServer(ctx, f.ShootClient.RESTConfig().CAData, f.ShootClient.RESTConfig().Host, tokenReq.Status.Token)
+				Expect(err).ToNot(HaveOccurred())
+
+				return status{
+					Code:    int(statusError.Code),
+					Message: statusError.Message,
+					Reason:  string(statusError.Reason),
+					Status:  statusError.Status,
+				}
+			}, time.Second*20, time.Second).Should(Equal(expectedStatus))
+
+			// Delete the oidc resource
+			err = f.ShootClient.Client().Delete(ctx, oidc, &client.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			// Expect API calls to be unauthorized
+			expectedStatus.Code = 401
+			expectedStatus.Message = "Unauthorized"
+			expectedStatus.Reason = string(metav1.StatusReasonUnauthorized)
+			Eventually(func() status {
+				statusError, err := requestAPIServer(ctx, f.ShootClient.RESTConfig().CAData, f.ShootClient.RESTConfig().Host, tokenReq.Status.Token)
+				Expect(err).ToNot(HaveOccurred())
+
+				return status{
+					Code:    int(statusError.Code),
+					Message: statusError.Message,
+					Reason:  string(statusError.Reason),
+					Status:  statusError.Status,
+				}
+			}, time.Second*20, time.Second).Should(Equal(expectedStatus))
+		} else {
+			f.Logger.Warning(fmt.Sprintf("Cannot register the seed as identity provider. Error: %s. The verification of oidc provider registration was skipped. Continuing with test execution...", err.Error()))
+		}
+
+		// Ensure that the OIDC service is disabled in order to verify the deletion process
+		f.UpdateShoot(ctx, ensureOIDCServiceIsDisabled)
+
+		// Ensure that oidc authenticator deployment is deleted
+		err = f.SeedClient.Client().Get(ctx, client.ObjectKeyFromObject(oidcDeployment), oidcDeployment)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsNotFound(err)).To(BeTrue())
```

Comment:
gardener have a matcher for this error
```suggestion
		Expect(err).To(BeNotFoundError()) // you have to import "github.com/gardener/gardener/pkg/utils/test/matchers"
```